### ISSUE TITLE
SDXL Dynamic prompt support

### ISSUE THREAD

**Issue by keturn** [invoke-ai/InvokeAI#4185]
When using Dynamic Prompts with SDXL:

- if you enter the `{ dynamic | part }` in the Style Prompt area, it is not expanded / replaced.
- if you enter the `{ dynamic | part }` in the Positive Prompt area, and you have the _Concatenate Prompt & Style_ link on, then it uses the replaced prompt for the _first_ encoder, but the second encoder receives the literal curly brackets with the full un-parsed text.

When Refiner is enabled, I think it also receives that unparsed version.

This bug is nasty, because initially the results of Dynamic Prompt make it _look_ like it's iterating though the various versions. But then when you go and try to replace your input with just one of those terms, the result is *different*, because the dynamic set was rendered with all the extra stuff in the second prompt.

### What version did you experience this issue on?

d09dfc3e9bdbd9a8522559fb8bd0bc70d205fd94

### Logs

```
>> [TOKENLOG] Tokens  (20):
a gumball machine full of { apples | pineapples | bananas }

>> [TOKENLOG] Tokens  (14):
a gumball machine full of pineapples

>> [TOKENLOG] Tokens  (20):
a gumball machine full of { apples | pineapples | bananas }
```

**Comment by keturn** [comment]
The other messy part of implementing this is going to be when both prompts contain dynamic elements.

Because if I have 
`dynamic(prompt_1="{ 🍒 | 🥝 } cola", prompt_2="", concat=True)`, I expect
- prompt_1="🍒 cola", prompt_2="🍒 cola"
- prompt_1="🥝 cola", prompt_2="🥝 cola"

but if I have 
`dynamic(prompt_1="{ 🍒 | 🥝 } cola", prompt_2="{ ⭐| 🔵 }", concat=True)`, I probably expect
- prompt_1="🍒 cola", prompt_2="⭐ 🍒 cola"
- prompt_1="🥝 cola", prompt_2="⭐ 🥝 cola"
- prompt_1="🍒 cola", prompt_2="🔵 🍒 cola"
- prompt_1="🥝 cola", prompt_2="🔵 🥝 cola"

but there's probably some other configuration where someone _does_ want to end up a set that includes
- prompt_1="🍒 cola", prompt_2="🥝 cola"
- prompt_1="🥝 cola", prompt_2="🍒 cola"
-- do we make them turn off concat as a way to indicate the expansions should be done independently?

**Comment by JPBarringer** [comment]
Not sure if this is a useful suggestion, but perhaps a warning message in the dynamic prompts pane when using SDXL models would be a good initial stopgap?

**Comment by TimCabbage** [comment]
It also happens when the dynamic parameters are cloned into the style field.
then, when generating the "normal" field receives the smaller, selected subset of tokens while the "style" field receives the unparsed "full" content.

**Comment by keturn** [comment]
hmm, _partially_ fixed with 3.2's queue.

If you _only_ use the Positive Prompt, then it works. The right thing goes to the Positive Prompt, and if you have style-prompt-concatenation on, it correctly gets copied through to the style prompt.

But anything put in the Style Prompt box doesn't have the Dynamic Prompts treatment applied to it.

**Comment by keturn** [comment]
(the easiest way to fix that is to banish the Style Prompt box from the main UI entirely and leave that level of prompt-hacking to the noders. but there are other options too I guess.)

**Comment by hipsterusername** [comment]
Fixed with 3.6 banishing of style prompt box.
